doc/ompl_constrained_planning/scripts/ompl_constrained_planning_tutorial.py

```diff
@@ -0,0 +1,736 @@
+            "/visualization_marker",
+            visualization_msgs.msg.Marker,
+            queue_size=20,
+        )
+        rospy.sleep(0.5)  # publisher needs some time to connect Rviz
+        self.remove_all_markers()
+        self.marker_id_counter = 0  # give each marker a unique idea
+
+        # Save some commenly used variables in the setup class
+        self.ref_link = self.move_group.get_pose_reference_frame()
+        self.ee_link = self.move_group.get_end_effector_link()
+        self.obstacle_name = "obstacle"
+
+    ## END_SUB_TUTORIAL
+
+    ## BEGIN_SUB_TUTORIAL start_state
+    ## Instead of using the current robot state as start state, we use a fixed state for the panda robot defined in the srdf config file.
+    ## The :code:`get_named_target_values` returns a dictionary with joint names and values for the "ready" position.
+    def create_start_state(self, named_target="ready"):
+        ready = self.move_group.get_named_target_values(named_target)
+
+        # Now create a robot state from these joint positions
+        joint_state = sensor_msgs.msg.JointState()
+        joint_state.header.stamp = rospy.Time.now()
+        joint_state.header.frame_id = self.move_group.get_pose_reference_frame()
+        joint_state.name = [key for key in ready.keys()]
+        joint_state.position = [val for val in ready.values()]
+
+        state = moveit_msgs.msg.RobotState()
+        state.joint_state = joint_state
+
+        return state
+
+    ## END_SUB_TUTORIAL
+
+    ## BEGIN_SUB_TUTORIAL pose_goal
+    ##
+    ## To keep things simple, we use the current end-effector pose to quickly create a reasonable goal.
+    ## We also visualize the start and goal position of the end-effector in Rviz with a simple sphere.
+    ## We assume that when the `demo.launch` file for the panda robot is launched, the robot is in the "ready" position.
+    def create_pose_goal(self):
+        self.move_group.clear_pose_targets()
+        pose = self.move_group.get_current_pose()
+
+        self.display_sphere(pose.pose, color=COLOR_RED)
+
+        pose.pose.position.y += 0.3
+        pose.pose.position.z -= 0.3
+
+        self.display_sphere(pose.pose)
+
+        return pose
+
+    ## For the second planning problem with the tilted plane, we need to create a pose goal that lies in this plane.
+    ## The plane is tilted by 45 degrees, so moving an equal amount in the y and z direction should be ok.
+    def create_pose_goal_in_plane(self):
+        self.move_group.clear_pose_targets()
+        pose = self.move_group.get_current_pose()
+
+        self.display_sphere(pose.pose, color=COLOR_RED)
+
+        pose.pose.position.x += 0.2
+        pose.pose.position.y += 0.3
+        pose.pose.position.z -= 0.3
+
+        self.display_sphere(pose.pose)
+
+        return pose
+
+    ## END_SUB_TUTORIAL
+
+    ## BEGIN_SUB_TUTORIAL ori_con
+    ## First we create simple box constraints on the current end-effector link (:code:`self.ee_link = "panda_link8"`).
```

Comment:
Change box constraints to orientation constraints.

Reply:
done

Reply:
I moved the orientation constraints definition after the position and Replaced "First" by "Then"